postpic/datahandling.py

```diff
@@ -386,14 +430,105 @@ def __copy__(self):
         ret.axes = [copy.copy(ret.axes[i]) for i in range(len(ret.axes))]
         return ret
 
+    # Stuff related with compatibility to Numpy's ufuncs starts here.
+
+    # make sure that np.array() * Field() returns a Field and not a plain array
+    __array_priority__ = 1
+
     def __array__(self, dtype=None):
         '''
         will be called by numpy function in case an numpy array is needed.
         '''
-        return np.asarray(self.matrix, dtype=dtype)
-
-    # make sure that np.array() * Field() returns a Field and not a plain array
-    __array_priority__ = 1
+        return np.asanyarray(self.matrix, dtype=dtype)
+
+    # What kind of other objects do we support? so far any kind of numpy array or scalar number
+    _HANDLED_TYPES = (np.ndarray, numbers.Number)
+
+    # handle ufuncs, new interface.
+    # see https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/[...]
+    # [...]/numpy.lib.mixins.NDArrayOperatorsMixin.html#numpy.lib.mixins.NDArrayOperatorsMixin
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        # Have implemented neither 'reduceat' because it is crazy nor
+        # 'inner' because it is not documented
+        if method not in ['__call__', 'reduce', 'outer', 'at', 'accumulate']:
+            return NotImplemented
+
+        out = kwargs.get('out', ())
+        for x in inputs + out:
+            # Only support operations with instances of _HANDLED_TYPES.
+            if not isinstance(x, self._HANDLED_TYPES + (type(self),)):
+                return NotImplemented
```

Comment:
* Can you add an error message, like "Type type(x) is not supported" or similar.
* Can you please add a line: `# TODO: add check of Axes extent and unit here`